Frontend/student-hub/src/pages/CoursePage.js

```diff
@@ -130,30 +178,33 @@ function CoursePage() {
         }
     };
 //like button
-    const handleLike = async (postId) => {
-        try {
-            const token = sessionStorage.getItem('token');
-            if (!token) {
-                navigate('/login');
-                return;
-            }
-            const response = await fetch(`http://localhost:8080/post/${postId}/like`, {
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/json',
-                    'Authorization': `Bearer ${token}`,
-                },
-            });
-            if (!response.ok) {
-                throw new Error('Failed to like post');
-            }
-            const newLike = await response.json();
-            setData(data.map(post => post.id === postId ? { ...post, Likes: [...post.Likes, newLike] } : post));
-            setFilteredData(filteredData.map(post => post.id === postId ? { ...post, Likes: [...post.Likes, newLike] } : post));
-        } catch (error) {
-            setError(error.message);
+const handleLike = async (postId) => {
+    try {
+        const token = sessionStorage.getItem('token');
+        if (!token) {
+            navigate('/login');
+            return;
         }
-    };
+
+        const response = await fetch(`${process.env.REACT_APP_API}/post/${postId}/like`, {
+
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${token}`,
+            },
+            body: JSON.stringify({ emojiId: selectedEmoji[postId] || '' }),
+        });
+        if (!response.ok) {
+            throw new Error('Failed to like post');
+        }
+        const newLike = await response.json();
+        setData(data.map(post => post.id === postId ? { ...post, Likes: [...post.Likes, newLike] } : post));
+        setFilteredData(filteredData.map(post => post.id === postId ? { ...post, Likes: [...post.Likes, newLike] } : post));
```

Comment:
`post => post.id === postId ? { ...post, Likes: [...post.Likes, newLike] } : post` is a bit repetitive here, it'd be great to split this to a helper method :)